lib/igniter.ex

```diff
@@ -183,57 +184,158 @@ if Code.ensure_loaded?(Igniter) do
       end
     end
 
+    def setup_application(igniter) do
```

Comment:
I created these two functions because it felt like there was a pattern. Looking at it afterwards, this all feels pretty phoenix specific with `Pheonix.Socket` and `Endpoint`s. 

I would refactor this and make these `defps` that are called as part of `setup_phoenix`. WDYT?